teps/0007-conditions-beta.md

```diff
@@ -0,0 +1,645 @@
+
+## Background
+
+### Conditions
+
+`Conditions` is a [custom resource](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) that enables users to specify a criteria to determine whether or not a `Task` executes. Its critical component is the `Check`, which specifies a `Step`, which is a `Container` that evaluates a `Condition`. If the container runs and exits with code 0, the `Condition` evaluates as `True`, otherwise it evaluates as `False`. With this design, users provide an image and Tekton runs it in a `Pod` to check the `Condition`, thus giving users flexibility to use whichever language. `Conditions` actually manifest themselves as `TaskRuns` that are handled differently from other `TaskRuns`. For example, when a `Condition` evaluates as `False`, the `TaskRun` from the `Condition` fails but it does not cause the whole `Pipeline` to fail, unlike the other `TaskRuns`.
+
+Users can specify `Conditions` and reference them in `Tasks` and `Pipelines` as such:
+```yaml
+apiVersion: tekton.dev/v1alpha1
+kind: Condition
+metadata:
+  name: file-exists
+spec:
+  params:
+    - name: 'path'
+  resources:
+    - name: workspace
+      type: git
+  check:
+    image: alpine
+    script: 'test -f $(resources.workspace.path)/$(params.path)'
+---
+apiVersion: tekton.dev/v1alpha1
+kind: Pipeline
+metadata:
+  name: conditional-pipeline
+spec:
+  resources:
+    - name: source-repo
+      type: git
+  params:
+    - name: path
+      default: 'README.md'
+  tasks:
+    - name: first-create-file
+      taskRef:
+        name: create-readme-file
+      resources:
+        outputs:
+          - name: workspace
+            resource: source-repo
+    - name: then-check
+      conditions:
+        - conditionRef: file-exists
+          params:
+            - name: path
+              value: '$(params.path)'
+          resources:
+            - name: workspace
+              resource: source-repo
+              from: [first-create-file]
+      taskRef:
+        name: echo-file-exists
+```
+For further information on `Conditions`, read the [documentation](https://github.com/tektoncd/pipeline/blob/master/docs/conditions.md), view [examples](https://github.com/tektoncd/pipeline/tree/master/examples) and review its uses in [Tekton plumbing](https://github.com/tektoncd/plumbing/blob/45b8b6f9f0bf10bf1cf462ee37597d1b44524fd8/tekton/ci/conditions.yaml). 
```

Comment:
nit: this link will be dead when we deprecate and remove Conditions (it's in a long time…), we may want to link to a released version of the doc.

Reply:
good point, will update the links